src/server/qgscapabilitiescache.cpp

```diff
@@ -52,12 +53,15 @@ const QDomDocument *QgsCapabilitiesCache::searchCapabilitiesDocument( const QStr
 
 void QgsCapabilitiesCache::insertCapabilitiesDocument( const QString &configFilePath, const QString &key, const QDomDocument *doc )
 {
-  if ( mCachedCapabilities.size() > 40 )
+  static const int cacheSize = qEnvironmentVariable( "QGIS_SERVER_CAPABILITIES_CACHE_SIZE", "40" ).toInt();
```

Comment:
You may want to use toInt( &ok ) and check if ok is true, raise a warning if not.